### ISSUE TITLE
Contact sync leads to a ton of empty "Organization" fields in contacts

### ISSUE THREAD

**Issue by jm355** [tutao/tutanota#7607]
- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Bug in mobile app**

**Describe the bug**
After enabling contact sync, many of my contacts suddenly had tons of empty "organization" fields in them.

**To Reproduce**
1. Enable system contact sync
2. Use phone for a while
3. open contacts, notice that some of them have tons of "organization" fields that are empty 

**Expected behavior**
No empty fields should be saved to contacts

**Smartphone (please complete the following information):**
 - Device: Pixel u
 - OS: GrapheneOS
 - Version v244.240912.0

**Additional context**
I had to disable contact syncing in tuta, export my contacts to a .vcf, then on my computer run `sed -i '/^ORG:\s*$/d' contacts.vcf`, delete all my contacts, and import the updated vcf. Even then I still see a bunch of contacts with the empty Organization fields

**Test Notes**
- [x] Login successfully in android device whose OS is GrapheneOs
- [x] Go to `settings -> contacts -> sync contacts`, Activate and deactivate `sync contacts` multiple times.
  - [ ] Go to native contacts in phone and open one of the contact. Only shows the Organization fields (company/role/department) if it has some values otherwise no empty Organization section.
  - [x] Go to native contacts in phone and manually link contacts. 
    - [x] If each linked contacts has Organization values, show those organization values as separate section. and no duplicated values.
    - [x]  If the Organization values are empty, then there is no empty Organization section

**Comment by wrdhub** [comment]
So, just to be clear: Are you seeing empty organization fields on the contacts in the Tuta app or on the contacts in your phone's contacts app (or both)?

**Comment by wrdhub** [comment]
Hey, I'm sorry but I feel like I do not fully understand the issue. Are you seeing these empty Organization fields in the contacts on the Tuta app or on your phone's contacts app?

I think some screenshots showing the issue (with private information redacted, of course) would be really helpful.

**Comment by jm355** [comment]
I'm seeing it in my phone contacts app, but not tuta 
![Screenshot_20240925-064445](https://github.com/user-attachments/assets/2e9f16c7-b7b7-49c7-8a2d-1a8be234311c)
![Screenshot_20240925-064454](https://github.com/user-attachments/assets/b8382826-bead-4908-a39a-af1dc8fcca5f)

**Comment by wrdhub** [comment]
We have had some other people complain about issues with our contact integration with GrapheneOS (#6568 ), so we are going to be looking into this more closely soon. Thanks again for the report.

**Comment by GoCho85** [comment]
Same goes for CalyxOS
Almost all contacts have like infinite organization fields
![Screenshot_20241003-103833_1](https://github.com/user-attachments/assets/a49b5fb6-17fd-46af-8edd-6599dc8a7e9c)
I could check the ones without the many fields, maybe they are different. 
What I first saw is, when I gave them an organization it only doubles the field.

**Comment by charlag** [comment]
We were unable to reproduce this issue so far. Would it be okay to ask you to create such a contact and then export it and share with us? This would help us greatly. Thanks.

**Comment by jm355** [comment]
Sure, I exported my contacts before and after enabling contact sync in tuta. Here's an excerpt from `diff -u contacts-before-tuta.vcf contacts-after-tuta.vcf`

```
BEGIN:VCARD
 VERSION:2.1
 N:Earth Customer Support;Magic;;;
 FN:Magic Earth Customer Support
 EMAIL:support@magicearth.com
+ORG:
+ORG:
+ORG:
+ORG:
+ORG:
 END:VCARD
 BEGIN:VCARD
 VERSION:2.1
 N:;Support;;;
 FN:Support
 EMAIL:support@ripstopbytheroll.com
+ORG:
+ORG:
+ORG:
+ORG:
+ORG:
 END:VCARD
```

and for clarity, here's that section straight out of contacts-before-tuta.vcf:
```
BEGIN:VCARD
VERSION:2.1
N:Earth Customer Support;Magic;;;
FN:Magic Earth Customer Support
EMAIL:support@magicearth.com
END:VCARD
BEGIN:VCARD
VERSION:2.1
N:;Support;;;
FN:Support
EMAIL:support@ripstopbytheroll.com
END:VCARD
```

and here's that section from contacts-after-tuta.vcf
```
BEGIN:VCARD
VERSION:2.1
N:Earth Customer Support;Magic;;;
FN:Magic Earth Customer Support
EMAIL:support@magicearth.com
ORG:
ORG:
ORG:
ORG:
ORG:
END:VCARD
BEGIN:VCARD
VERSION:2.1
N:;Support;;;
FN:Support
EMAIL:support@ripstopbytheroll.com
ORG:
ORG:
ORG:
ORG:
ORG:
END:VCARD
```

**Comment by charlag** [comment]
thank you for the info, we will try to see where it's coming from

**Comment by jm355** [comment]
I created a contact in Tuta 252.241122.0, and right off the bat it has one empty Organization field. Looking at the other examples I shared, they now have way more empty organization fields than they used to.

**Comment by jm355** [comment]
I the Tuta app, I don't see any organization fields

**Comment by GoCho85** [comment]
I testet with 252.241122.0 and deleted every contact on my phone before that. I don't find any additional company fields besides the one I create myself.
So it seems to work for me

**Comment by jm355** [comment]
Did you create the contact in the tuta app? So far the test contacts I made haven't gotten more org fields, but they do still have one empty one

**Comment by GoCho85** [comment]
Uhm I exported all existing contacts from the app to my local device. 
But you are right, even the contacts without a "organization" note have the field. So I guess a little progress, because we don't have endless organizations

**Comment by jm355** [comment]
Yup. It would also be nice if tuta handled cleaning up the contacts that already have a ton of empty org fields.

**Comment by charlag** [comment]
I'm closing this as it seems like this does not happen anymore (it was probably contact scheme mismatch with some contact apps).

**Comment by hrb-hub** [comment]
Reopening this issue as it still occurs, at least on Graphene OS.

**Comment by parachute6326** [comment]
I have the same issue with many organization fields in the default contacts app on GrapheneOS. If I deactivate and reactivate the synchronization in the Tuta app, only one organization field is displayed in the default contacts app of GrapheneOs, even if there is no entry in the organization field. But after some time, every contact without an entry in the organization field gets more and more organization fields. For contacts with an entry in the organization field, however, there is only one organization field.
(Sorry, I am not a native English speaker)

**Comment by BijinDev** [comment]
@parachute6326 Could you also mention the app version you are using.

**Comment by parachute6326** [comment]
I'm using version v299.250722.0 of the Tuta app.

**Comment by hrb-hub** [comment]
A couple of things I found when testing:
- When the `Department` field is set but both `Company` and `Role` are empty, the `Organization` field isn't shown in the native contacts app. See #9900 
- There's still a way to duplicate the organization field and show an empty `Organization` by doing the following (though this appears to be an existing issue, see #9901):
  - Have contact sync enabled.
  - On Tuta:
    - Set a `Department` but keep `Company` and `Role` empty, then save the contact.
    - Without changing `Department`, set `Role` and save.
    - Without changing `Department`, remove `Role` and save.
    - Remove `Department`.
  - Check the contact using the native contacts app, and see that the contact now has 3 `Organization` fields (1x with department + role, 1x with department, 1x empty).
